### PR TITLE
nitweb: list linearized definitions

### PR DESCRIPTION
**Pull request by Morriar**
Display linearization lists:

* Mclass linearize all the class definitions
   * linearization of `core::Array` mclassdefs: http://nitweb.moz-code.org/class/core::Array

* MProperty linearize all the prop definitions
    * linearization of `core::Object::hash`: http://nitweb.moz-code.org/property/core::Object::hash

* MClassDef linearize to intro
    * linearization of `core::flat$Array`: http://nitweb.moz-code.org/classdef/core::flat$Array

* MPropdef linearize to intro
    * linearization of `core$Array$Object::SELF`: http://nitweb.moz-code.org/propdef/core$Array$Object::SELF

### CODE REVIEW

File: src/nitweb.nit [hunk 1]
```diff
@@ -100,6 +100,7 @@ class APIRouter
 		use("/entity/:id", new APIEntity(model, mainmodule))
 		use("/code/:id", new APIEntityCode(model, mainmodule, modelbuilder))
 		use("/uml/:id", new APIEntityUML(model, mainmodule))
+		use("/linearization/:id", new APIEntityLinearization(model, mainmodule))
```
Comment (privat) [inline src/nitweb.nit]: i'm not fan of the *linearization* word. Maybe use `redef` to remember the keyord?

File: src/web/model_api.nit [hunk 2]
```diff
@@ -120,6 +120,60 @@ class APIEntity
...
+		if mentity == null then return
+		var arr = new JsonArray
+		if mentity isa MClass then
+			arr.add_all mclass_lin(mentity)
+		else if mentity isa MClassDef then
+			arr.add_all mclassdef_lin(mentity)
+		else if mentity isa MProperty then
+			arr.add_all mproperty_lin(mentity)
+		else if mentity isa MPropDef then
+			arr.add_all mpropdef_lin(mentity)
```
Comment (privat) [inline src/web/model_api.nit]:
~~~
var lin = mentity.lin_for_doc
if lin == null then
  res = 404 (Do not know how to linearize such an entity)
  return
end
~~~

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
Reroll: libified the `MEntity::collect_linearization`

**Pull request merged**